ThumbnailSequence.cpp

```diff
@@ -1560,7 +1560,7 @@ ThumbnailSequence::CompositeItem::CompositeItem(
 bool
 ThumbnailSequence::CompositeItem::incompleteThumbnail() const
 {
-	return dynamic_cast<IncompleteThumbnail*>(m_pThumb) != 0;
+	return m_pThumb == 0 && dynamic_cast<IncompleteThumbnail*>(m_pThumb) != 0;
```

Comment:
Where exactly does it crash?

It shouldn't be crashing in dynamic_cast, as dynamic_cast()ing a null pointer is legal:
http://stackoverflow.com/questions/5155820/portably-safe-to-pass-null-zero-to-dynamic-cast